Tasks/Src/RTOS_BPS.c

```diff
@@ -38,9 +40,33 @@ void RTOS_BPS_MutexPost(BPS_OS_MUTEX* mutex, BPS_OS_OPT options) {
     OSMutexPost(mutex, options, &err);
     assertOSError(err);
 }
-void RTOS_BPS_TaskCreate(void){
 
-}
+/**
+ * @brief   Creates a task that will be handled by the OS
+ * @param   *p_tcb - pointer to the tcb
+ * @param   *p_name - pointer to task name
+ * @param   p_task - pointer to the task
+ * @param   *p_args - pointer to task function arguments
+ * @param   *prio - task priority
+ * @param   *p_stk_base - the stack
+ * @param   stk_size - size of the stack
+ * @param   *p_err - return error code
+ * @return  nothing to see here
+ */
+void RTOS_BPS_TaskCreate(
+    OS_TCB        *p_tcb,
+    CPU_CHAR      *p_name,
+    OS_TASK_PTR    p_task,
+    void          *p_arg,
+    OS_PRIO        prio,
+    CPU_STK       *p_stk_base,
+    CPU_STK_SIZE   stk_size,
+    OS_ERR        *p_err)
+    {
```

Comment:
To add onto this @Champers5000 if you find the source code for Micrium's TaskCreate, you will see a lot of these don't really need special types. TASK_NAME is just a char pointer, TASK_PTR is just a function pointer, etc. 

This func seems hard to abstract out because of how specific Micrium requires these things. FreeRTOS is a bit better - https://www.freertos.org/a00125.html.
This function will be rewritten when we port to FreeRTOS, anyways.